sds/src/scanner/mod.rs

```diff
@@ -286,6 +315,58 @@ impl Scanner {
         output_rule_matches
     }
 
+    #[cfg(feature = "match_validation")]
+    pub async fn validate_matches(
+        &self,
+        rule_matches: &mut Vec<RuleMatch>,
+    ) -> Result<(), MatchValidationError> {
+        if !self.scanner_features.return_matches {
+            return Err(MatchValidationError::NoMatchValidationType);
+        }
+        // Create MatchValidatorRuleMatch per match_validator_type to pass it to each match_validator
+        let mut match_validator_rule_match_per_type = AHashMap::new();
+        for rule_match in rule_matches.drain(..) {
```

Comment:
I'm not familiar with the drain method. Why is it needed in this case compare to an iter?